hadoop-tools/hadoop-azure/src/main/java/org/apache/hadoop/fs/azurebfs/services/AbfsClient.java

```diff
@@ -783,6 +784,12 @@ public AbfsRestOperation append(final String path, final byte[] buffer,
         abfsUriQueryBuilder.addQuery(QUERY_PARAM_CLOSE, TRUE);
       }
     }
+    if (reqParams.isRetryDueToExpect()) {
```

Comment:
we can assign correct header in the first place? Lets prevent replace. Also, what does `EXPECT=""` as header can do maybe unknown.